data-products/src/main/scala/org/sunbird/analytics/job/report/CourseMetricsJob.scala

```diff
@@ -191,65 +213,60 @@ object CourseMetricsJob extends optional.Application with IJob with ReportGenera
-    val resolvedExternalIdDF = userBlockResolvedDF.join(externalIdMapDF, Seq("userid"), "left_outer")
+    val userStateResolvedDF = userBlockResolvedDF.join(resolvedStateDenormDF, Seq("userid"), "left_outer")
 
     /*
     * Resolve organisation name from `rootorgid`
     * */
 
-    val resolvedOrgNameDF = resolvedExternalIdDF
-      .join(organisationDF, organisationDF.col("id") === resolvedExternalIdDF.col("rootorgid"), "left_outer")
-      .select(resolvedExternalIdDF.col("userid"), resolvedExternalIdDF.col("rootorgid"), col("orgname").as("orgname_resolved"))
+    val resolvedOrgNameDF = userStateResolvedDF
+      .join(organisationDF, organisationDF.col("id") === userStateResolvedDF.col("rootorgid"), "left_outer")
+      .select(userStateResolvedDF.col("userid"), userStateResolvedDF.col("rootorgid"), col("orgname").as("orgname_resolved"))
 
-    /*
-    * Resolve school name from `orgid`
-    * */
-    val schoolNameDF = resolvedExternalIdDF
-      .join(organisationDF, organisationDF.col("id") === resolvedExternalIdDF.col("organisationid"), "left_outer")
-      .select(resolvedExternalIdDF.col("userid"), resolvedExternalIdDF.col("organisationid"), col("orgname").as("schoolname_resolved"))
-
-    /**
-     * If the user present in the multiple organisation, then zip all the org names.
-     * Example:
-     * FROM:
-     * +-------+------------------------------------+
-     * |userid |orgname                             |
-     * +-------+------------------------------------+
-     * |user030|SACRED HEART(B)PS,TIRUVARANGAM      |
-     * |user030| MPPS BAYYARAM                     |
-     * |user001| MPPS BAYYARAM                     |
-     * +-------+------------------------------------+
-     * TO:
-     * +-------+-------------------------------------------------------+
-     * |userid |orgname                                                |
-     * +-------+-------------------------------------------------------+
-     * |user030|[ SACRED HEART(B)PS,TIRUVARANGAM, MPPS BAYYARAM ]      |
-     * |user001| MPPS BAYYARAM                                         |
-     * +-------+-------------------------------------------------------+
-     * Zipping the orgnames of particular userid
-     *
-     *
-     */
-    val schoolNameIndexDF = schoolNameDF.withColumn("index", count("userid").over(Window.partitionBy("userid").orderBy("userid")).cast("int"))
-
-    val resolvedSchoolNameDF = schoolNameIndexDF.selectExpr("*").filter(col("index") === 1).drop("organisationid", "index", "batchid")
-      .union(schoolNameIndexDF.filter(col("index") =!= 1).groupBy("userid").agg(collect_list("schoolname_resolved").cast("string").as("schoolname_resolved")))
-
-    resolvedExternalIdDF
-      .join(resolvedSchoolNameDF, Seq("userid"), "left_outer")
+    val finalReportDF = userStateResolvedDF
+      .join(userInfoDF, Seq("userid"), "left_outer")
       .join(resolvedOrgNameDF, Seq("userid", "rootorgid"), "left_outer")
       .dropDuplicates("userid")
```

Comment:
Do we still have duplicates at this point?

Reply:
Duplicate records are coming if we remove the drop duplicate from this part of the code

Reply:
Which code is causing the duplicates?